pkg/kubelet/cm/devicemanager/pod_devices.go

```diff
@@ -321,6 +323,20 @@ func (pdev *podDevices) deviceRunContainerOptions(podUID, contName string) *Devi
 			annotationsMap[k] = v
 			opts.Annotations = append(opts.Annotations, kubecontainer.Annotation{Name: k, Value: v})
 		}
+
+		// Updates for CDIDevices
+		for _, cdiDevice := range resp.CdiDevices {
+			if _, ok := cdiDevicesMap[cdiDevice.Name]; ok {
+				klog.V(4).InfoS("Skip existing CDI Device", "name", cdiDevice.Name)
+				continue
+			}
```

Comment:
Is cdiDeviceMap update anywhere? If not, how can it contain existing CDI devices?

Reply:
It's updated on line 334 of this change set. It was missing off the first revision though.

Reply:
it makes sense to me now, thanks.